File: front-end/src/pedidos/DTO_PedidoCompleto.ts
import { Pedido } from './pedido';
import { PedidoProduto } from 'src/pedidoProduto/pedidoProduto';

export class DTO_PedidoCompleto{
    Pedido: Pedido;
    PedidoProduto: PedidoProduto[];
}

File: back-end/BlueModas.Domain/DTOs/DTO_PedidoCompleto.cs
using BlueModas.Domain.Entities;
using System;
using System.Collections.Generic;
using System.Text;

namespace BlueModas.Domain.DTOs
{
    public class DTO_PedidoCompleto
    {
        public Pedido Pedido { get; set; }
        public IEnumerable<PedidoProduto> PedidoProduto { get; set; }
    }
}
